spring-integration-http/src/main/java/org/springframework/integration/http/inbound/HttpRequestHandlingEndpointSupport.java

```diff
@@ -505,7 +503,23 @@ protected void onInit() throws Exception {
 
 			Message<?> reply = null;
 			if (this.expectReply) {
-				reply = this.sendAndReceiveMessage(message);
+				try {
+					reply = this.sendAndReceiveMessage(message);
+				}
+				catch (MessageTimeoutException e) {
+					if (this.statusCodeExpression != null) {
+						reply = getMessageBuilderFactory().withPayload(e.getMessage())
```

Comment:
Can't we really reply it as a `ErrorMessage` with that `StatusCode` header of course.
Seems for me we can return `Exception` for the response as is and it will be converted properly...